ethcore/src/externalities.rs

```diff
@@ -116,6 +116,10 @@ impl<'a, T: 'a, V: 'a, B: 'a> Ext for Externalities<'a, T, V, B>
 	where T: Tracer, V: VMTracer, B: StateBackend
 {
 	fn initial_storage_at(&self, key: &H256) -> vm::Result<H256> {
+		if self.substate.new_contract_with_previously_non_empty_storage.contains(&self.origin_info.address) {
```

Comment:
Is it possible that the contract was empty initially, but then it is somehow initialized via `CREATE` within the same scope?
I.e. asking if returning `zero()` here wouldn't break something else

Reply:
I changed this to use `is_base_storage_root_unchanged`. If base storage root is changed from an original storage root value, then the changed base storage root can only be `KECCAK_NULL_RLP` (because we only change it in contract creation). From any point after that, we discard the actual original storage value, but use `H256::zero()`.

I think it's safe and can't think of any way this will break things, but would love to have more eyes for review.